src/java.naming/share/classes/com/sun/jndi/toolkit/url/Uri.java

```diff
@@ -364,7 +364,7 @@ private void parseCompat(String uri) throws MalformedURLException {
                             + (port == -1?"":(":" + port));
                     if (!auth.equals(hostport)) {
                         // throw if we have user info or regname
-                        throw new MalformedURLException("unsupported authority: " + auth);
+                        throw new MalformedURLException("Authority component is not server-based, or contains user info. Unsupported authority: " + auth);
                     }
```

Comment:
This looks okay but you may have to split up the line to avoid adding a 150+ char line (most of the file seems to keep the lines under 100 or so).